compiler/rustc_session/src/session.rs

```diff
@@ -427,7 +427,11 @@ impl Session {
     {
         let old_count = self.err_count();
         let result = f();
-        if self.err_count() == old_count { Ok(result) } else { Err(ErrorReported) }
+        if self.err_count() == old_count {
+            Ok(result)
+        } else {
+            Err(ErrorReported::new_i_know_what_im_doing())
```

Comment:
`err_count` can be something like `Option<(NonZeroUsize, ErrorReported)>`, making this only slightly harder than simply `==`.

Reply:
Tried this, but it creates a lot of noise everywhere else that `err_count` is used...